lib/dry/operation/extensions/active_record.rb

```diff
@@ -109,8 +111,17 @@ def included(klass)
               #   @see Dry::Operation#steps
               #   @see https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/DatabaseStatements.html#method-i-transaction
               klass.define_method(:transaction) do |connection = default_connection, **opts, &steps|
-                connection.transaction(**options.merge(opts)) do
-                  intercepting_failure(-> { raise ::ActiveRecord::Rollback }, &steps)
+                intercepting_failure(method(:throw_failure)) do
```

Comment:
Looking at the two extensions we're changing here, we're passing `method(:throw_failure)` as the first argument in both cases.

Would you expect this `throw_failure` method to be used like this in all extensions?

If so, I wonder if we could make it the default. I toyed with changing the `#intercepting_failure` params to:

```ruby
def intercepting_failure(handler = method(:throw_failure), &block)
```

And then updating these extensions, so all they do is `intercepting_failure do`.

With these, the tests all still pass. What do you think?

Reply:
That's a great suggestion. A smart change to make it look much cleaner. Done!